packages/orchestration/src/exos/portfolio-holder-kit.js

```diff
@@ -0,0 +1,200 @@
+  return zone.exoClassKit(
+    'PortfolioHolderKit',
+    {
+      invitationMakers: M.interface('InvitationMakers', {
+        Action: M.call(ChainNameM, M.string(), M.arrayOf(M.any())).returns(
+          VowShape,
+        ),
+      }),
+      holder: M.interface('Holder', {
+        asContinuingOffer: M.call().returns({
+          publicSubscribers: TopicsRecordShape,
+          invitationMakers: M.any(),
+        }),
+        getPublicTopics: M.call().returns(TopicsRecordShape),
+        getAccount: M.call(ChainNameM).returns(M.remotable()),
+        addAccount: M.call(
+          ChainNameM,
+          M.remotable(),
+          PublicTopicShape,
+        ).returns(),
+      }),
+      getInvitationMakersWatcher: M.interface('GetInvitationMakersWatcher', {
+        onFulfilled: M.call(
+          M.splitRecord({
+            invitationMakers: M.any(),
+            publicSubscribers: M.any(),
+          }),
+        )
+          .optional({
+            action: M.string(),
+            invitationArgs: M.arrayOf(M.any()),
+          })
+          .returns(M.promise()),
+      }),
+    },
+    /**
+     * @param {Iterable<[string, OrchestrationAccount<any>]>
+     *   | CopyMap<string, import('@endo/marshal').Passable>} accountEntires
+     * @param {Iterable<[string, ResolvedPublicTopic<unknown>]>
+     *   | CopyMap<string, import('@endo/marshal').Passable>} publicTopicEntires
+     */
+    (accountEntires, publicTopicEntires) => {
+      mustMatch(accountEntires, AccountEntriesShape, 'must provide accounts');
+      mustMatch(
+        publicTopicEntires,
+        PublicTopicsShape,
+        'must provide public topics',
+      );
+      const accounts = harden(
+        makeScalarBigMapStore('accounts', { durable: true }),
+      );
+      const publicTopics = harden(
+        makeScalarBigMapStore('publicTopics', { durable: true }),
+      );
+      accounts.addAll(accountEntires);
+      publicTopics.addAll(publicTopicEntires);
+      return /** @type {PortfolioHolderState} */ (
+        harden({
+          accounts,
+          publicTopics,
+        })
+      );
+    },
+    {
+      invitationMakers: {
+        /**
+         * @param {string} chainName key where the account is stored
+         * @param {string} action invitation maker name, e.g. 'Delegate'
+         * @param {unknown[]} invitationArgs
+         */
+        Action(chainName, action, invitationArgs) {
+          const { accounts } = this.state;
+          accounts.has(chainName) || Fail`no account found for ${chainName}`;
+          const account = accounts.get(chainName);
+          return watch(
+            E(account).asContinuingOffer(),
+            this.facets.getInvitationMakersWatcher,
+            { action, invitationArgs },
+          );
+        },
+      },
+      holder: {
+        // XXX type. like `OrchestrationAccountI['asContinuingOffer']` but
+        // returns a Vow<Invitation> instead of a Promise<Invitation> as the
+        // accounts are remote
```

Comment:
```suggestion
        /** @type {HostOf<OrchestrationAccountI['asContinuingOffer']>} */
```
That gives an error because this isn't returning a Vow. Mustn't it?

Reply:
This path is tested in multichain-testing, and appears to work. The caller of `makeAccountsInvitation` from `auto-stake-it` gets this in their offer result and we can see public topics in vstorage.
We can add another step to the multichain test that calls `Undelegate` to demonstrate the invitationMakers are callable.

Reply:
Glad it works. However `asContinuingOffer` in `OrchestrationAccountI` says it returns a Promise. So I think we should have this function return a Vow. But I don't see a Vow-equivalent of `Promise.resolve()`. 

Alternately we could change `asContinuingOffer` to return `ERef` so both sync and Promise are valid. 

Either of those will need more work which is out of scope of this PR. So please leave a FIXME but also please include `/** @type {HostOf<OrchestrationAccountI['asContinuingOffer']>} */` so it's clear what is supposed to work.

Reply:
>I don't see a Vow-equivalent of Promise.resolve()

`asVow`?

Reply:
I've decided to follow the interface of `OrchestrationAccountI` and am now returning Vows here. Please re-review `PortfolioHolder` in the next pass as it's changed a bit.

There is a lingering type issue I've marked in the same place of OP - it seems to not like the `invitationMakers` type and I'm coming up short on a fix so far.



> asVow?

It's an export from `vowTools`. There are a few PRs: https://github.com/Agoric/agoric-sdk/pulls?q=asVow

These might be of interest: https://github.com/Agoric/agoric-sdk/pull/9577, https://github.com/Agoric/agoric-sdk/pull/9696